subprojects/test-kit/src/main/java/org/gradle/testkit/runner/internal/DefaultGradleRunner.java

```diff
@@ -60,6 +69,10 @@ public DefaultGradleRunner() {
         this.gradleExecutor = gradleExecutor;
         this.testKitDirProvider = testKitDirProvider;
         this.debug = Boolean.getBoolean(DEBUG_SYS_PROP);
+
+        if (!JavaVersion.current().isJava8Compatible()) {
```

Comment:
:x: TestKit uses the Tooling API under the covers. The deprecations is probably not needed here as the check is already part of `DefaultConnection.java`.